### DISCUSSION TITLE
Selecting points multiple times in one go

### DISCUSSION THREAD

**Discussion by venugovh** [pyvista/pyvista#1886] [Q&A] [answered]
Hi,

I am using the `enable_point_picking()` method to select points on a mesh. However, I need to select single points at multiple locations in one go instead of having to plot and select different points every time. I cannot use the `enable_cell_picking()` method because the point locations are scattered. A dragging window would include unnecessary points.

Is there a way to tailor this method or is there another solution so that once I plot the mesh with the points, I can select multiple points and return them as a `NumPy `array?

Thank you! This library is a lifesaver :)

**Comment by banesullivan** [comment, marked as answer]
Totally possible. use the `callback` argument of `enable_point_picking()` to track your picked points. Something like:

```py
import pyvista as pv
from pyvista import examples

mesh = examples.download_bunny()

da_points = []
p = pv.Plotter(notebook=0)
p.add_mesh(mesh)

def callback(pt):
    da_points.append(pt)
    p.add_point_labels([pt,], ['You picked me!'], )

p.enable_point_picking(callback)
p.show()
```

https://user-images.githubusercontent.com/22067021/143785784-b6b70470-54b4-40a1-97e7-256304825443.mov

**Reply by venugovh** [reply]
Would the same functions work for `pyvistaqt.BackgroundPlotter()`?

Thank you!

**Reply by banesullivan** [reply]
it should!
